Assets/Scripts/InventoryMenu.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityStandardAssets.Characters.FirstPerson;
+using TMPro;
+
+public class InventoryMenu : MonoBehaviour
+{
+    [SerializeField]
+    private GameObject inventoryMenuItemTogglePrefab;
+    [SerializeField]
+    private Transform inventoryListContentArea;
+    [SerializeField]
+    private TextMeshProUGUI itemLabelText;
+    [SerializeField]
+    private TextMeshProUGUI itemDescriptionAreaText;
+
+    private static InventoryMenu instance;
+    public static InventoryMenu Instance
+    {
+        get
+        {
+            if (instance == null)
+            {
+                throw new System.Exception("There is currently no InventoryMenu instance." +
+                    " Make sure an InventoryMenu script is applied to a GameObject.");
+            }
+            return instance;
+        }
+        private set { instance = value; }
+    }
+
+    private AudioSource audioSource;
+    private CanvasGroup canvasGroup;
+    private RigidbodyFirstPersonController rigidbodyFirstPersonController;
+
+    private bool IsVisible => canvasGroup.alpha > 0;
+
+    private void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else
+        {
+            throw new System.Exception("There is already an instance of InventoryMenu.");
+        }
+
+        audioSource = GetComponent<AudioSource>();
+        canvasGroup = GetComponent<CanvasGroup>();
+        rigidbodyFirstPersonController = FindObjectOfType<RigidbodyFirstPersonController>();
+    }
+
+    private void Start()
+    {
+        HideMenu();
+        StartCoroutine(WaitForAudioClip());
+    }
+
+    private void Update()
+    {
+        if (Input.GetButtonDown("Inventory"))
+        {
+            if (IsVisible)
```

Comment:
It would look a bit cleaner here if you got rid of the { } in these if statements since they each only have one line of code. But that's more personal preference, and if you consistently do it this way, I think it's fine the way it is.